apps/builder/app/builder/features/sidebar-left/panels/marketplace/utils.ts

```diff
@@ -0,0 +1,87 @@
+import { useStore } from "@nanostores/react";
+import { atom } from "nanostores";
+import { loadProjectDataById, type Data } from "@webstudio-is/http-client";
+import {
+  extractWebstudioFragment,
+  findTargetAndInserеFragment,
+} from "~/shared/instance-utils";
+import type { WebstudioData } from "@webstudio-is/sdk";
+import productsData from "./products.json";
+import { MarketplaceProduct } from "./schema";
+
+const $activeProductId = atom<MarketplaceProduct["id"] | undefined>();
+
+export const $activeProductData = atom<WebstudioData | undefined>();
+
+export const $products = atom(
+  productsData.map((product, index) =>
+    MarketplaceProduct.parse({ ...product, id: String(index) })
+  )
+);
+
+export const categories: Array<{
+  category: MarketplaceProduct["category"];
+  label: string;
+}> = [{ category: "templates", label: "Templates" }];
+
+export const useActiveProduct = () => {
+  const activeProductId = useStore($activeProductId);
+  const products = useStore($products);
+  const product = activeProductId
+    ? products.find((product) => product.id === activeProductId)
+    : undefined;
+
+  const setActiveProduct = async (id?: MarketplaceProduct["id"]) => {
+    $activeProductId.set(id);
+    const product = products.find((product) => product.id === id);
+    if (product === undefined || id === undefined) {
+      return;
+    }
+    let webstudioData;
+    try {
+      const data = await loadProjectDataById({
```

Comment:
should probably rewrite to use a remix route and get the error handling out of the box, show an error instead of results